sharktank/sharktank/types/tensors.py

```diff
@@ -1171,6 +1211,8 @@ def add_to_archive(self, builder: ShardedArchiveBuilder) -> InferenceTensorMetad
             {"": self.name},
             extra_properties={
                 "shard_count": len(self._shards),
+                "devices": self.devices,
```

Comment:
Nope, we shouldn't be putting this in the archive. Pipeline parallelism won't have any importance in comparison to sharding needing to know count/orientation.